[ADD] emc_website_portal: Cooperator Information

remytms opened the pull request:
This module show the cooperator information on the website portal.

It lets you access to your : 
- Release Capital Requests
- Cooperator Certificate
- Cooperator information (bank account, cooperator register number, number of shares,…)

remytms committed:
[ADD] emc_wsite_portal: Cooperator information

Add the cooperator information on the /my/home page.

remytms committed:
[ADD] emc_wsite_portal: Release Capital Requests

Add the access to the release capital request in the personal portal
view.

remytms committed:
[ADD] emc_wsite_portal: Cooperator Certificate

Add a button that allow to download the cooperator certificate.

There is some access right issue for now. So for the moment the function
`send_cooperator_certificate()` in the controller is not used.

remytms committed:
[FIX] emc_wsite_portal: Access right issue

remytms committed:
[FIX] emc_website_portal: PDF download issue

The bug was that trying to download the cooperator certificate or
another PDF in the interface brings an error 500. This was due to a
wrong usage of the function `get_pdf()` and also due to a error of
encoding when using `format()` python function with `{name}`. Instead I
use `"%s" % name` that works great.

It didn't use `get_pdf()` right because of an old version of mis-builder
that cause the methord `get_pdf()` to work weird. With an updated
version of mis-builder, `get_pdf()` can be used correctly.

remytms committed:
[FIX] emc_wsite_portal: Show followed invoices

The issue was that the portal shows all the release capital request that
are followed by the user (generally because, it's him that that accepted
the request). The page should only show the capital request of that the
user owns.

remytms committed:
[FIX] emc_wsite_portal: Access right cap request

For people who are not in group_portal they cannot access to the list of
their capital request.

remytms commented:
A review is welcome. :)

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2015-2016 Odoo S.A.
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
```
remove other contributors who didn't contributed

remytms replied:
I have put these Copyright because the function `_get_archive_groups()` is copied from their code (website_portal_v10 and website_portal).

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+        response = super(CooperatorWebsiteAccount, self).account()
+        partner = request.env.user.partner_id
+
+        invoice_mgr = request.env['account.invoice'].sudo()
```
sudo() should go in function call

remytms replied:
Ok, I will change it. :)

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+        """
+        values = self._prepare_portal_layout_values()
+        partner = request.env.user.partner_id
+        invoice_mgr = request.env['account.invoice'].sudo()
```
sudo should go in function calls

remytms replied:
ok

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+        # is not allowed to access to it or if the object is not found.
+        partner = request.env.user.partner_id
+        invoice_mgr = request.env['account.invoice']
+        capital_request = invoice_mgr.sudo().browse(oid)
```
good boy ;)

remytms replied:
^^

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+        except MissingError:
+            raise NotFound()
+        # Get the pdf
+        report_mgr = request.env['report'].sudo()
```
sudo should go in function call

remytms replied:
ok

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+    def get_cooperator_certificat(self, **kw):
+        """Render the cooperator certificate pdf of the current user"""
+        partner = request.env.user.partner_id
+        report_mgr = request.env['report'].sudo()
```
again sudo()...

remytms replied:
ok

houssine78 commented on easy_my_coop_website_portal/controllers/main.py:
```diff
@@ -0,0 +1,187 @@
+    def _get_archive_groups_sudo(self, model, domain=None, fields=None,
+                                 groupby="create_date",
+                                 order="create_date desc"):
+        """Same as the one from website_portal_v10 except that it runs
```
not possible to call the _get_archive_groups of website_portal_v10 with sudo() instruction?
something like _get_archive_groups.sudo() ?

remytms replied:
I have tried at the beginning but this function comes from a controller which is not part of a `recordset` and therefor has no member `sudo()`. I have tried to change the become `sudo` by changing the user in the context but the current context but it doesn't work either. If someone has, a better way to do this, I will take it. But for now it's the only way that works.

houssine78 replied:
ok

houssine78 commented on easy_my_coop_website_portal/views/easy_my_coop_website_portal_templates.xml:
```diff
@@ -0,0 +1,203 @@
+     Copyright 2015-2016 Odoo S.A.
+     Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
remove non contributor of this module

remytms replied:
I have put their names because of the template `portal_my_home_menu_capital_release` that comes from their code in website_portal_v10.

houssine78 commented on easy_my_coop_website_portal/views/easy_my_coop_website_portal_templates.xml:
```diff
@@ -0,0 +1,203 @@
+        <p t-if="coop.company_register_number">
+          <label>Company Register Number: </label>
+          <t t-esc="coop.company_register_number"/>
+        </p>
```
this should be displayed only if coop.is_company

remytms replied:
You're right.

houssine78 commented on easy_my_coop_website_portal/views/easy_my_coop_website_portal_templates.xml:
```diff
@@ -0,0 +1,203 @@
+        </p>
+        <p t-if="coop.representative">
+          <label>Legal Representative: </label>
+          <t t-esc="coop.representative"/>
```
this should be displayed only if coop.is_company

remytms replied:
ok.

houssine78 commented:
by the way there is no translation files at least we need it for french BE, FR and CH and for dutch BE

remytms committed:
[FIX] emc_wsite_portal: Explicit usage of sudo

When using sudo, write it explicitly before each query.
Prevent from showing company information if the cooperator is not a
company.
